### PR TITLE
feat(spdm-pal): add set_dpe_only_slot() for DPE-only certificate chains

### PR DESCRIPTION
**Pull request by yujieh-mrvl**
Add a new `set_dpe_only_slot()` method to `SharedCertStore` that configures a slot with an empty endorsement chain and `RootHash = SHA-384(first DPE cert)`. This enables SPDM certificate chains starting directly from LDevID without static Root CA or IDevID certs, for deployments where the verifier obtains trust anchors out-of-band. Unit tests included.

Related: #2010

### CODE REVIEW

**Comment by yujieh-mrvl** [comment]
@amit-kumarh Could you please review this PR when you have a chance? Thanks!

**Review by amit-kumarh — COMMENTED** [review]
This should be rebased on top of #2061 which adds postquantum certificate chain support. Please run `cargo fmt` as well to satisfy the precheckin. Otherwise, LGTM. Thanks!

File: runtime/userspace/api/spdm-lib/pal/src/cert/store.rs [hunk 1]
```diff
@@ -102,6 +102,47 @@ impl SharedCertStore {
...
+        if idx >= NUM_CERT_SLOTS || ecc_first_dpe_cert.is_empty() {
+            return Err(mcu_error::codes::INVARIANT);
+        }
+        static EMPTY_CHAIN: &[&[u8]] = &[];
+
+        let ecc_hash = compute_root_hash(alloc, ecc_first_dpe_cert).await?;
+        let mut ro = ReadOnlyEndorsement::new(EMPTY_CHAIN, ecc_hash);
+
+        if let Some(cert) = mldsa_first_dpe_cert {
+            if !cert.is_empty() {
```
Comment (amit-kumarh) [inline runtime/userspace/api/spdm-lib/pal/src/cert/store.rs:133]: nit: We could consider making this noisier, to catch a platform accidentally passing an empty buffer as an MLDSA cert instead of silently falling back to ECC only

**Review by amit-kumarh — APPROVED** [review]
LGTM. Can you please authorize the CLA as well to allow merging this? @yujieh-mrvl

File: runtime/userspace/api/spdm-lib/pal/src/cert/store.rs (hunk 1 above)
Comment (yujieh-mrvl) [reply runtime/userspace/api/spdm-lib/pal/src/cert/store.rs:133]: Great catch, thanks! You're right — silently falling back to ECC-only means a platform that fumbles its ML-DSA cert (e.g. passes an uninitialized/zero-length buffer) would ship a slot that quietly only serves ECC, which is exactly the kind of bug that's painful to notice in the field.

**I'll make the contract explicit and fail loudly:**

- `None` → the platform intentionally opts out of an ML-DSA chain (the only valid way to "skip")
- `Some(non-empty)` → serve ML-DSA
- `Some(empty)` → treated as a programming error, returns `INVARIANT` instead of degrading to ECC-only

Concretely, I'm pulling the input checks into a small **synchronous** helper called at the very top of `set_dpe_only_slot()`, before any allocation/hashing. This fails fast and makes it unit-testable without an async executor (this crate has no `block_on`/executor dev-dependency, and the `Some(empty)` path would otherwise sit behind the ECC `compute_root_hash().await`):

```rust
fn validate_dpe_only_inputs(
    idx: usize,
    ecc_first_dpe_cert: &[u8],
    mldsa_first_dpe_cert: Option<&[u8]>,
) -> McuResult<()> {
    if idx >= NUM_CERT_SLOTS || ecc_first_dpe_cert.is_empty() {
        return Err(mcu_error::codes::INVARIANT);
    }
    if matches!(mldsa_first_dpe_cert, Some(c) if c.is_empty()) {
        return Err(mcu_error::codes::INVARIANT);
    }
    Ok(())
}
```

`set_dpe_only_slot()` then calls `validate_dpe_only_inputs(...)?` first, and the ML-DSA branch drops its inner emptiness check (non-empty is now guaranteed).

I'll also:
- update the doc comment to state the `Some(empty)` → `INVARIANT` contract
- add a test pinning it: `Some(empty)` → `Err`; `None` / `Some(non-empty)` → `Ok` (plus the existing empty-ECC / out-of-range cases)
Does this approach look good to you? I'll push the change once you confirm.
